src/agent/src/image_rpc.rs

```diff
@@ -37,73 +38,58 @@ impl ImageService {
         Self { sandbox }
     }
 
-    fn build_oci_path(cid: &str) -> PathBuf {
-        let mut oci_path = PathBuf::from("/tmp");
-        oci_path.push(cid);
-        oci_path.push(IMAGE_OCI);
-        oci_path
-    }
-
-    fn pull_image_from_registry(image: &str, cid: &str, source_creds: &Option<&str>) -> Result<()> {
+    fn pull_image_from_registry(
+        image: &str,
+        cid: &str,
+        source_creds: &Option<&str>,
+        policy_path: &Option<&String>,
+    ) -> Result<()> {
         let source_image = format!("{}{}", "docker://", image);
 
-        let mut manifest_path = PathBuf::from("/tmp");
-        manifest_path.push(cid);
-        manifest_path.push("image_manifest");
-        let target_path_manifest = format!("dir://{}", manifest_path.to_string_lossy());
-
-        // Define the target transport and path for the OCI image, without signature
-        let oci_path = Self::build_oci_path(cid);
+        let tmp_cid_path = Path::new("/tmp/").join(cid);
+        let oci_path = tmp_cid_path.join(IMAGE_OCI);
         let target_path_oci = format!("oci://{}", oci_path.to_string_lossy());
 
-        fs::create_dir_all(&manifest_path)?;
         fs::create_dir_all(&oci_path)?;
 
         info!(sl!(), "Attempting to pull image {}...", &source_image);
 
         let mut pull_command = Command::new(SKOPEO_PATH);
         pull_command
-            // TODO: need to create a proper policy
-            .arg("--insecure-policy")
             .arg("copy")
             .arg(source_image)
-            .arg(&target_path_manifest);
+            .arg(&target_path_oci)
+            .arg("--remove-signatures");
 
+        // If source credentials were passed (so not using an anonymous registry), pass them through
         if let Some(source_creds) = source_creds {
             pull_command.arg("--src-creds").arg(source_creds);
         }
 
-        let status: ExitStatus = pull_command.status()?;
-        ensure!(
-            status.success(),
-            "failed to copy image manifest: {:?}",
-            status,
-        );
-
-        // Copy image from one local file-system to another
-        // Resulting image is still stored in manifest format, but no longer includes the signature
-        // The image with a signature can then be unpacked into a bundle
-        let status: ExitStatus = Command::new(SKOPEO_PATH)
-            .arg("--insecure-policy")
-            .arg("copy")
-            .arg(&target_path_manifest)
-            .arg(&target_path_oci)
-            .arg("--remove-signatures")
-            .status()?;
+        // If a policy_path provided, use it, otherwise fall back to allow all image registries
+        if let Some(policy_path) = policy_path {
+            pull_command.arg("--policy").arg(policy_path);
+        } else {
+            pull_command.arg("--insecure-policy");
+        }
 
-        ensure!(status.success(), "failed to copy image oci: {:?}", status);
+        let status: ExitStatus = pull_command.status()?;
```

Comment:
I think it would be worth adding a `debug!()` call to log `pull_command` so the admin gets to see the full CLI.

Reply:
This should be done now.